components/suggestion/channel_mention_provider.jsx

```diff
@@ -139,7 +140,10 @@ export default class ChannelMentionProvider extends Provider {
             });
         });
         wrappedChannels = wrappedChannels.sort((a, b) => {
-            return ChannelUtils.sortChannelsByDisplayName(a.channel, b.channel);
+            //
+            // MM-12677 When this is migrated this needs to be fixed to pull the user's locale
+            //
+            return sortChannelsByTypeAndDisplayName('en', a.channel, b.channel);
```

Comment:
These changes are going to conflict with my LocalizationStore PR since I just changed channel_utils to use the redux store directly